src/main/java/jenkins/plugins/http_request/auth/FormAuthentication.java

```diff
@@ -71,5 +71,10 @@ public FormValidation doCheckKeyName(@QueryParameter String value) {
         public String getDisplayName() {
             return "Form Authentication";
         }
+
+        public String getHttpRequestHelpPath() {
+            return "/descriptor/"+HttpRequest.class.getName()+"/help";
```

Comment:
This is complicated and caused me a headache. The original plugin was hardcoding paths to the help files with code like:
```
help="/plugin/http_request/WEB-INF/classes/jenkins/plugins/http_request/HttpRequest/help-keyName.html"
```
When I created the `HttpRequestGlobalConfig.java` and the `HttpRequestStep.java`, I wanted to leave the help files under `src/main/resources/jenkins/plugins/http_request/HttpRequest`, because the help files are mostly the same between the Pipeline (snippet generator) and the traditional plugin (build step). For sharing the help files between the classes, I used code like this
```
help="${descriptor.httpRequestHelpPath}/url"
```
So instead of having the two forms to refer to the help files, I unified every help file reference to use the later form.